File: AtCoder/ACP/BootCamp4b/Medium100/python/C_Buy_an_Integer.py
MAX_N = 1_000_000_000
A, B, X = list(map(int, input().split()))
M = len(str(min(X, MAX_N)))
res = 0
for i in range(1, M + 1):
    t = (X - B * i) // A
    if 1 <= t and i <= len(str(t)):
        res = max(res, min(t, MAX_N if i == len(str(MAX_N)) else 10 ** i - 1))
print(res)

File: AtCoder/ACP/BootCamp4b/Medium100/cpp/C_Buy_an_Integer.cpp
#include <algorithm>
#include <iostream>
#include <math.h>
#include <stdio.h>
#include <string>

#define MAX_N 1'000'000'000

typedef long long ll;

using namespace std;

int main() {
  ll A, B, X;
  cin >> A >> B >> X;
  int M = to_string(min(X, (ll)MAX_N)).size();
  ll res = 0;
  for (int i = M; i > 0; --i) {
    ll t = (X - B * i) / A;
    if (1 <= t && i <= to_string(t).size()) {
      res = max(res, min(t, i == to_string(MAX_N).size() ? (ll)MAX_N : (ll)pow(10, i) - 1));
    }
  }
  printf("%lld\n", res);
}